### ISSUE TITLE
iOS Safari nav exapanding over zoom controls

### ISSUE THREAD

**Issue by SamHall** [openseadragon/openseadragon#919]
Unlike other mobile browsers iOS Safari's normal behavior is to shrink the top address bar and hide the bottom navigation during fullscreen zoom. In many situations - if one touches the top address bar area the top and bottom bar expand - and remain that way - but **hides** the built-in controls.

This leaves only one option for the user - to reload the whole page to get out of the zoomed state.
On an iPhone 6 Plus - the hidden issue occurs upon zoom itself vs. the accidental touching of the top bar.  Very frustrating - and I know it's more of a Safari issue - but hope someone here may have insight into a workaround.

I use Bootstrap.  The issue occurs if I use the openseadragon-canvas in less than a full width grid (example:  have content on the same row in a different column).
And I've seen it manifest itself on other sites.  

Any suggestions would be greatly appreciated!

![giphy](https://cloud.githubusercontent.com/assets/4950023/14626108/a7cc1cbc-05b6-11e6-8c1d-4844308711af.gif)

Example page:  [http://deepzoomchattanooga.com/gpn2.php](http://deepzoomchattanooga.com/gpn2.php)

Code:
```
<!DOCTYPE html>
<html lang="en">
  <head>
    <!-- Remove When Live -->
    <META NAME="ROBOTS" CONTENT="NOINDEX, FOLLOW">
      <META NAME="ROBOTS" CONTENT="INDEX, NOFOLLOW">
        <META NAME="ROBOTS" CONTENT="NOINDEX, NOFOLLOW">
          <!-- Remove When Live -->
          <meta charset="utf-8">
          <meta http-equiv="X-UA-Compatible" content="IE=edge">
          <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
          <meta name="apple-mobile-web-app-capable" content="yes">
          <meta name="mobile-web-app-capable" content="yes">
          <!-- Opengraph Data -->
          <title>
            New Page
          </title>
          <!-- Bootstrap core CSS -->
          <link href="/bootstrap336/css/theme.min.css" rel="stylesheet">
          <!-- IE10 viewport hack for Surface/desktop Windows 8 bug -->
          <link href="/bootstrap336/css/ie10-viewport-bug-workaround.css" rel="stylesheet">
          <!-- Custom styles for this template -->
          <style>
            .openseadragon {
              height: 568px;
              background-color: #000;
            }
            p {
              padding: 10px 0;
            }
          </style>
          <!-- Fav and touch icons -->
          </head>
        <body>
          <div class="container">   
            <!-- Fixed navbar -->
            <!-- Begin page content -->
            <div class="row">
              <div class="col-md-12">
                <h1 class="heading text-center">Demo
                  <br />
                  <small>
                    Set One
                  </small>
                </h1>
                <p>
                  Lorem ipsum dolor sit amet, ad eam epicuri adipiscing delicatissimi, ludus menandri eu mel, has cu dolorum graecis neglegentur. Eum an populo facilisi, no solum soluta iriure has, sint tractatos interesset et sit. Vix ea appetere scriptorem, audire abhorreant eum no. Amet recusabo sententiae per ex. Ut sea velit ornatus facilis. Ignota regione cu sea, apeirian iudicabit ut eam.
                </p>
              </div>
            </div>
            <div class="row">  
              <div class="col-md-6 col-xs-12">
                <div class="openseadragon" id="openseadragon-canvas1">
                </div>
                <p>
                  Lorem ipsum dolor sit amet, ad eam epicuri adipiscing delicatissimi, ludus menandri eu mel, has cu dolorum graecis neglegentur. Eum an populo facilisi, no solum soluta iriure has, sint tractatos interesset et sit. Vix ea appetere scriptorem, audire abhorreant eum no. Amet recusabo sententiae per ex. Ut sea velit ornatus facilis. Ignota regione cu sea, apeirian iudicabit ut eam.
                </p>
              </div>
              <div class="col-md-6 col-xs-12">
<h1 class="jumbotron text-center">
Zoom Images for everyone</h1>
                <div class="openseadragon" id="openseadragon-canvas2">
                </div>

              </div>
            </div>
          </div>
        </body>
        <!-- Placed at the end of the document so the pages load faster -->
        <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min.js">
        </script>
        <script src="/bootstrap336/js/bootstrap.min.js">
        </script>
        <!-- IE10 viewport hack for Surface/desktop Windows 8 bug -->
        <script src="/bootstrap336/js/ie10-viewport-bug-workaround.js">
        </script>
        <script src="openseadragon-bin-2.1.0/openseadragon.min.js">
        </script>
        <!-- GPN ZOOM -->		
        <script type="text/javascript">
          var viewer1 = new OpenSeadragon.Viewer({
            id: "openseadragon-canvas1",
            prefixUrl: "openseadragon-bin-2.1.0/images/",
            tileSources: [
              "/zoom/glassplates/gpn1/dzc_output.xml",
            ],
          }
                                                );
        </script>
        <!-- GPN ZOOM -->			
        <script type="text/javascript">
          var viewer2 = new OpenSeadragon.Viewer({
            id: "openseadragon-canvas2",
            prefixUrl: "openseadragon-bin-2.1.0/images/",
            tileSources: [
              "/zoom/glassplates/gpn2/dzc_output.xml",
            ],
          }
                                                );
        </script>
        </html>

```

**Comment by iangilman** [comment]
Playing with your example on an iPhone, there is another way for the user to get the controls back: tapping at the very top of the address bar. This is the standard gesture for scrolling to the top of the page, and sure enough it scrolls the image down enough that the controls become visible again. 

Of course we want to do this programmatically instead of relying on the user to do it. Perhaps you could watch for resize and/or scroll events to see if there's a way to know when the address bar appears, and then resize and/or scroll the viewer or page. 

If you find a good solution for your page, maybe there's something from that that could be included into OpenSeadragon to make it even more automatic. 

Incidentally, since zoom and pan are such common gestures on smart phones, the user doesn't really need any of the buttons in the OSD control bar except for the "exit fullscreen" button. Maybe you could take advantage of that fact somehow.

**Comment by SamHall** [comment]
Well I must admit I had not tried that.  Apparently this is an iOS Safari issue only that is driving many developers nuts.
I typically use custom buttons and hide unwanted buttons via options.  (I re-enabled those on the demo link now)

However,  I believe hiding or changing what is shown on fullpage requires additional script - yes?
I do realize this isn't really an OpenSeadragon issue as much as it is an Apple/Safari pain.
Thanks.

**Comment by iangilman** [comment]
Did you have any luck watching for size/scroll events on the window?

And yes, if you want to do something special in full page mode, you can watch for OSD's events for going into and out of that mode.

**Comment by SamHall** [comment]
Not yet - had to focus on boring full-time job stuff : )
But I will revisit the subject / solution.

Thanks.
